### PR TITLE
improve(finalizer): Smarter filtering for Optimism transactions

### PR DESCRIPTION
**Pull request by nicholaspai**
Reduces RPC requests sent to Optimism by reducing number of messages that are very likely out of proof/finalization window

Improves logging

### CODE REVIEW

File: src/finalizer/index.ts [hunk 1]
```diff
@@ -131,21 +133,22 @@ export async function finalize(
...
       const crossChainMessenger = getOptimismClient(chainId, hubSigner) as optimismSDK.CrossChainMessenger;
-      const firstBlockToFinalize = await getBlockForTimestamp(
+
+      // First submit proofs for any newly withdrawn tokens. You can submit proofs for any withdrawals that have been
+      // snapshotted on L1, so it takes roughly 1 hour from the withdrawal time. Skip events older than 2 days old.
+      const oldestBlockToProve = await getBlockForTimestamp(
         hubPoolClient.chainId,
         chainId,
-        getCurrentTime() - optimisticRollupFinalizationWindow,
+        getCurrentTime() - 2 * oneDaySeconds,
```
Comment (pxrl) [inline src/finalizer/index.ts:142]: Optimism write here that block rates are 2 seconds now, but can be subject to change. Probably a good idea to note that in the comment above?

https://community.optimism.io/docs/developers/bedrock/differences/#block-production

File: src/finalizer/index.ts [hunk 2]
```diff
@@ -93,6 +93,7 @@ export async function finalize(
         at: "Finalizer",
         message: `Oldest TokensBridged block to attempt to finalize for ${getNetworkName(chainId)}`,
         firstBlockToFinalize,
+        firstBlockSecondsAgo: optimisticRollupFinalizationWindow,
```
Comment (pxrl) [inline src/finalizer/index.ts:96]: I might be misunderstanding this, but I'm confused about the name `firstBlockSecondsAgo`. optimisticRollupFinalizationWindow defaults to 5 days (line 53); `firstBlockSecondsAgo` indicates to me that we're finalising blocks from that point up to head.

File: src/finalizer/index.ts [hunk 3]
```diff
@@ -131,21 +133,22 @@ export async function finalize(
...
       );
-
-      // First submit proofs for any newly withdrawn tokens. You can submit proofs for any withdrawals that have been
-      // snapshotted on L1, so it takes roughly 1 hour from the withdrawal time. Skip events older than 7 days old.
       logger.debug({
         at: "Finalizer",
         message: `Earliest TokensBridged block to attempt to submit proofs for ${getNetworkName(chainId)}`,
-        earliestBlockToProve: firstBlockToFinalize,
+        oldestBlockToProve,
+        oldestBlockSecondsAgo: 2 * oneDaySeconds,
```
Comment (james-a-morris) [inline src/finalizer/index.ts:149]: Is `oneDaySeconds` meant to represent one day in seconds? The variable naming is a tad confusing.

**Review by james-a-morris — APPROVED** [review]
LGTM - just a minor nit about the variable name

**Pull request closed without merging**